src-web/components/kappnav/ComponentView.jsx

```diff
@@ -17,19 +17,83 @@
  *****************************************************************/
 
 import 'carbon-components/scss/globals/scss/styles.scss';
-import React from 'react'
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
 import lodash from 'lodash'
 import { Module, ModuleBody } from 'carbon-addons-cloud-react'
 import { Loading } from 'carbon-components-react'
 import ResourceTable from './common/ResourceTable.js'
 import StructuredListModule from './common/StructuredListModule'
-import { CONTEXT_PATH, PAGE_SIZES, SORT_DIRECTION_ASCENDING } from '../../actions/constants'
+import { CONTEXT_PATH, PAGE_SIZES, SORT_DIRECTION_ASCENDING, RESOURCE_TYPES } from '../../actions/constants'
 import { getRowSlice, sort, sortColumn } from '../../actions/common'
+import SecondaryHeader from './common/SecondaryHeader.jsx'
 import getResourceData, { refreshResource, refreshResourceComponent } from '../../definitions/index'
 import msgs from '../../../nls/kappnav.properties'
 
-class ComponentView extends React.Component {
-  constructor (props){
+class ComponentView extends Component {
+
+  /**
+   * @return {RESOURCE_TYPES} the current resource type 
+   */
+  getResourceType() {
+    if(location.pathname === '/') {
+      // There is no url path, error case
+      return undefined
+    }
+    
+    let tokenized_url = this.getTokenizedURLPath()
+
+    // Assuming there is always an index 1 because ComponentView is only 
+    // used with URLs that contain the resourcetype at index 1
+    const resourceTypeInURL = tokenized_url[1].toLowerCase()
+    if(resourceTypeInURL === 'applications') {
+      return RESOURCE_TYPES.APPLICATION
+    } else if (resourceTypeInURL === 'was-nd-cells') {
+      return RESOURCE_TYPES.WASNDCELL
+    } else {
+      return RESOURCE_TYPES.LIBERTYCOLLECTIVE
+    }
```

Comment:
I do not think we should have references to `was` or `liberty` specific resources in kAppNav